### PR TITLE
Enable linux and win hostonly devicelab tests.

### PR DESCRIPTION
**Pull request by godofredoc**
## Description

This PR moves most of the hostonly devicelab tests to luci except for 6
that are still failing in LUCI.

## Related Issues

Bug:
  https://github.com/flutter/flutter/issues/63242

## Tests

N/A

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [X] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by godofredoc** [comment]
All the hostonly_devicelab tests are now running separately. Linux and mac are all passing, I kept the ones failing in windows to continue running them on cirrus while we fix them.

**Review by christopherfujino — COMMENTED** [review]
It looks like you ran dartfmt on this `dev/bots/test.dart`, if you revert that it would make the diff smaller and easier to review. It would also make a regression easier to spot. The framework intentionally does not use `dartfmt`, see https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting

**Comment by godofredoc** [comment]
> It looks like you ran dartfmt on this `dev/bots/test.dart`, if you revert that it would make the diff smaller and easier to review. It would also make a regression easier to spot. The framework intentionally does not use `dartfmt`, see https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting

Reverted back the formatting. I'll check if I still get the error missing {} after ifs.

**Comment by christopherfujino** [comment]
> > It looks like you ran dartfmt on this `dev/bots/test.dart`, if you revert that it would make the diff smaller and easier to review. It would also make a regression easier to spot. The framework intentionally does not use `dartfmt`, see https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting
> 
> Reverted back the formatting. I'll check if I still get the error missing {} after ifs.

looks like analysis passed :)

File: .cirrus.yml [hunk 1]
```diff
@@ -238,7 +238,7 @@ task:
       <<: *FIREBASE_SHARD_TEMPLATE
 
     - name: web_smoke_test
-      only_if: "changesInclude('.cirrus.yml', 'examples/hello_world/**' ,'dev/**', 'packages/flutter/**', 'packages/flutter_test/**', 'packages/flutter_tools/lib/src/test/**', 'packages/flutter_web_plugins/**', 'bin/**') || $CIRRUS_PR == ''"
+      only_if: "changesInclude('.cirrus.yml', 'examples/hello_world/**' ,'dev/**', 'packages/flutter/**', 'packages/flutter_test/**', 'packages/flutter_tools/lib/src/test/**', 'packages/flutter_web_plugins/**', 'bin/**') || $CIRRUS_PR != ''"
```
Comment (christopherfujino) [inline .cirrus.yml:241]: these diffs are always so hard for me to read...

So this is saying we always want to run the test for any PR, and on post-submit only on affected diffs...right?

This doesn't seem like the right logic. Should it be `&& $CIRRUS_PR == ''`, which would be only run pre-submit with a diff on the affected files?

Comment (godofredoc) [reply .cirrus.yml:241]: I see, yeah I think it has to be && but I thought $CIRRUS_PR == '' meant postsubmit.

Comment (christopherfujino) [reply .cirrus.yml:241]: lol, yeah you're right. so `&& $CIRRUS_PR != ''`

Comment (christopherfujino) [reply .cirrus.yml:241]: This is why I put question marks at the end of all my statements, I never get this right the first try.

Comment (godofredoc) [reply .cirrus.yml]: yeah, errors in those conditionals are usually difficult to spot. Updated.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**